cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java

```diff
@@ -864,4 +871,27 @@ protected Future<Void> deletePersistentClaims() {
                             .deletePersistentClaims(maybeDeletePvcs, desiredPvcs);
                 });
     }
+
+    /**
+     * Defers to the Kafka metadata state manager if there is a KRaft migration rollback ongoing and in such case,
+     * it will delete the /controller znode to allow brokers to elect a new controller among them, now that KRaft
+     * controllers are out of the picture.
+     *
+     * @return  Completes when the possible /controller znode deletion is done
+     */
+    protected Future<Void> maybeDeleteControllerZnode() {
+        return ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
+                .compose(compositeFuture -> {
+                    String zkConnectionString = KafkaResources.zookeeperServiceName(reconciliation.name()) + ":" + ZookeeperCluster.CLIENT_TLS_PORT;
+                    LOGGER.infoCr(reconciliation, "Maybe deleting /controller znode on {}", zkConnectionString);
+                    this.kafkaMetadataStateManager.maybeDeleteZooKeeperControllerZnode(
+                            reconciliation,
+                            compositeFuture.resultAt(0),
+                            compositeFuture.resultAt(1),
+                            operationTimeoutMs,
+                            zkConnectionString
+                    );
+                    return Future.succeededFuture();
+                });
```

Comment:
This should be conditioned by something and not run every single reconciliation.